src/game_objects/game_object.cpp

```diff
@@ -8,16 +9,48 @@ GameObject::GameObject() {
     id = id_counter++;
 }
 
-// Renders model in scene
+GameObject::~GameObject(){
+    for (auto it = children.begin(); it != children.end(); it++) {
+        delete(*it);
+    }
+}
+
+// Adopt a game object
+void GameObject::add_child(GameObject *obj) {
+    children.push_back(obj);
+}
+
+// Disown a single child
+void GameObject::remove_child(GameObject *obj) {
+    // I'm pretty sure this line of code works. Not tested.
+    children.erase(std::remove(children.begin(), children.end(), obj), children.end());
```

Comment:
Please test if there's any doubt. Helps to remove problems with people using your code since they can better assume that the bugs are coming from new code and not previous code.